code/datums/components/rot.dm

```diff
@@ -14,14 +35,9 @@
 	var/atom/A = parent
 
 	var/turf/open/T = get_turf(A)
-	if(!istype(T))
-		return
 
-	var/datum/gas_mixture/stank = new
-	ADD_GAS(/datum/gas/miasma, stank.gases)
-	stank.gases[/datum/gas/miasma][MOLES] = amount
-	T.assume_air(stank)
-	T.air_update_turf()
+	if(istype(T) && !istype(T,/turf/open/space))
+		GLOB.MIASMA_PROCESSOR.tilestodiffuse[T] += amount
```

Comment:
This is causing runtimes.
```
[16:22:59] Runtime in rot.dm,40: Cannot read null.tilestodiffuse
  proc name: process (/datum/component/rot/process)
  src: /datum/component/rot/gibs (/datum/component/rot/gibs)
  call stack:
  /datum/component/rot/gibs (/datum/component/rot/gibs): process(10)
  Processing (/datum/controller/subsystem/processing): fire(0)
  Processing (/datum/controller/subsystem/processing): ignite(0)
  Master (/datum/controller/master): RunQueue()
  Master (/datum/controller/master): Loop()
  Master (/datum/controller/master): StartProcessing(0)
```

Reply:
so basically, it isn't fully initialized and glob_datum doesn't do that. how do I go about to initialize it?